test/unit/gateways/checkout_v2_test.rb

```diff
@@ -482,6 +482,46 @@ def test_successful_authorize_and_capture_with_additional_options
     assert_success capture
   end
 
+  def test_successful_authorize_and_capture_with_additional_options_ammount_allocation
+    response = stub_comms(@gateway, :ssl_request) do
+      options = {
+        card_on_file: true,
+        transaction_indicator: 2,
+        previous_charge_id: 'pay_123',
+        processing_channel_id: 'pc_123',
+        amount_allocations: [
```

Comment:
You might want tests for when these are missing the `id` or `amount`. I'm assuming there are tests already where this isn't included.